server/src/main/scala/org/ensime/server/DocsHandler.scala

```diff
@@ -0,0 +1,98 @@
+// Copyright: 2010 - 2016 https://github.com/ensime/ensime-server/graphs
+// License: http://www.gnu.org/licenses/gpl-3.0.en.html
+package org.ensime.server
+
+import java.nio.file.{ Files, Paths }
+
+import io.netty.buffer.{ ByteBuf, Unpooled }
+import io.netty.channel.{ ChannelFuture, ChannelFutureListener, ChannelHandlerContext, SimpleChannelInboundHandler }
+import io.netty.handler.codec.http.HttpHeaderNames.CONTENT_TYPE
+import io.netty.handler.codec.http.HttpMethod.GET
+import io.netty.handler.codec.http.HttpResponseStatus
+import io.netty.handler.codec.http.HttpResponseStatus.{ BAD_REQUEST, FORBIDDEN, NOT_FOUND, OK }
+import io.netty.handler.codec.http.HttpVersion.HTTP_1_1
+import io.netty.handler.codec.http.{ DefaultFullHttpResponse, FullHttpRequest, HttpUtil }
+import io.netty.util.CharsetUtil
+
+import org.ensime.core.DocJarReading
+
+class DocsHandler(docs: DocJarReading) extends SimpleChannelInboundHandler[FullHttpRequest] {
+
+  val docRegex = """/docs/([^/]+\.jar)/(.+)""".r
```

Comment:
aside: if you use a capital on a regex value, you can use it in the pattern matcher more easily